### PR TITLE
Restrict end date for prep date range

### PR DESCRIPTION
**Pull request by conbrad**
- Adapts and attributes code from https://github.com/jungsoft/materialui-daterange-picker
- Limits the end date to 7 days after the start date in the UI date range picker
- Adds reset button to clear the selected range
- Removes `materialui-daterange-picker` dependency
- Adds `date-fns` dependency for handling native JavaScript `Date`s: https://date-fns.org 
# Test Links:
[Percentile Calculator](https://wps-pr-1818.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1818.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1818.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1818.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1818.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1818.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1818.apps.silver.devops.gov.bc.ca/hfi-calculator)
[FWI Calculator](https://wps-pr-1818.apps.silver.devops.gov.bc.ca/fwi-calculator)

### CODE REVIEW

**Review by andrea-williams — CHANGES_REQUESTED** [review]
There's a problem when going back in time. E.g., select a prep period for August 2021. Once that data is retrieved, try setting a prep period for this week - it seems you can't??? (I haven't been able to figure out how to get around it anyway).

**Comment by conbrad** [comment]
> There's a problem when going back in time. E.g., select a prep period for August 2021. Once that data is retrieved, try setting a prep period for this week - it seems you can't??? (I haven't been able to figure out how to get around it anyway).

Yeah I need to fix that. Need a way to reset the start / end dates so that the range isn't restricted

**Comment by conbrad** [comment]
> There's a problem when going back in time. E.g., select a prep period for August 2021. Once that data is retrieved, try setting a prep period for this week - it seems you can't??? (I haven't been able to figure out how to get around it anyway).

Just fixed a change that adds a button to reset the date range

**Review by tessmgood — APPROVED** [review]

File: web/src/components/dateRangePicker/DateRangePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+const DateRangePicker: React.FunctionComponent<DateRangePickerProps> = (
+  props: DateRangePickerProps
+) => {
+  const today = new Date()
+
+  const { open, onChange, initialDateRange, maxDate } = props
+
+  const minDateValid = addYears(today, -10)
+  const maxDateValid = parseOptionalDate(maxDate, addYears(today, 10))
+  const [intialFirstMonth, initialSecondMonth] = getValidatedMonths(
```
Comment (andrea-williams) [inline web/src/components/dateRangePicker/DateRangePicker.tsx:41]:
```suggestion
  const [initialFirstMonth, initialSecondMonth] = getValidatedMonths(
```

**Review by andrea-williams — COMMENTED** [review]
Codecov is reporting code smells that I'd like to see addresses/formally dismissed before you merge

**Comment by conbrad** [comment]
> Codecov is reporting code smells that I'd like to see addresses/formally dismissed before you merge

Addressed, they were from original source code.

**Review by andrea-williams — APPROVED** [review]

**Review by JulianForeman — APPROVED** [review]

**Pull request merged**